### ISSUE TITLE
Crash on when adding new realm object at Xcode 10 Official Release

### ISSUE THREAD

**Issue by thihaaung6245** [realm/realm-cocoa#5923]
!!! MANDATORY TO FILL OUT !!!

## Goals
To work perfectly with Xcode 10 which mean it was working before with Xcode 9.4.1

## Expected Results
Crash
<img width="1421" alt="screenshot at sep 19 17-05-59" src="https://user-images.githubusercontent.com/10526402/45748600-46ac9380-bc2f-11e8-822d-b506b5242f10.png">
<img width="1420" alt="screenshot at sep 19 17-07-14" src="https://user-images.githubusercontent.com/10526402/45748602-47452a00-bc2f-11e8-8365-a785c58158bf.png">

## Steps to Reproduce
It just crash after the data is writing to realm database.

## Version of Realm and Tooling

Realm framework version: 3.9.0

Xcode version: 10

iOS/OSX version: 12

Dependency manager + version: Cocoapods

**Comment by mkhoshpour** [comment]
same problem here. some of my models is being saved and one of them gets crashed here.

**Comment by santiguillen** [comment]
> same problem here. some of my models is being saved and one of them gets crashed here.

me too, on Xcode 9 working

**Comment by tgoyne** [comment]
There's not much I can do to help you with just a screenshot of a crash.

**Comment by neoplastic** [comment]
I'm seeing the same issue but only when running my app on iOS 9.x. 10.x and above is alright.

**Comment by ggirotto** [comment]
@neoplastic Exactly, my problem was solved identically with what you said, but in my case, I had a var `var customGroupedBy: String?` that shouldn't be read by Realm, it must just be an object property. To solve that I added this property in `ignoredProperties` function from Realm and this worked flawlessly

**Comment by tgoyne** [comment]
Thanks for the additional info; I can now reproduce this.

**Comment by tgoyne** [comment]
Looks like this is actually just #5781, and the fix for that (#5783) was incorrect and only worked by coincidence.

**Comment by thihaaung6245** [comment]
Thanks everyone and @tgoyne . Please let me know when this fixes was live and I will close the issue. Right now setting as non-optional is the only way.
